python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1585,17 +1439,10 @@ def _execute_asset_backfill_iteration_inner(
         dynamic_partitions_store=instance_queryer,
     )
 
-    if request_roots:
-        check.invariant(
-            len(run_requests) > 0,
-            "At least one run should be requested on first backfill iteration",
-        )
-
     updated_asset_backfill_data = AssetBackfillData(
         target_subset=asset_backfill_data.target_subset,
         latest_storage_id=next_latest_storage_id or asset_backfill_data.latest_storage_id,
-        requested_runs_for_target_roots=asset_backfill_data.requested_runs_for_target_roots
-        or request_roots,
+        requested_runs_for_target_roots=True,
```

Comment:
trying to think through if this change makes rollbacks any dicier.

This particular thing could have just always been True before right? Since we always requested them on the first iteration?

Reply:
yeah that's right -- also regardless of that bit, if we did an iteration of the backfill with this new logic, then we certainly would have requested all of the target roots regardless of what the old logic was